Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -652,26 +652,27 @@ public class CsharpCompiler : SinglePassCompiler {
       }
 
       string PrintInvocation(Formal f, int i) {
-        var prefix = customReceiver
-          ? datatype.IsRecordType || !f.HasName ? $"(({DtCtorDeclarationName(ctor)}{typeArgs}) _this)." : "_this.dtor_"
-          : "";
-        var (name, type) = ($"{prefix}{FormalName(f, i)}", f.Type);
+        var name = customReceiver
+          ? datatype.IsRecordType || !f.HasName
+            ? $"(({DtCtorDeclarationName(ctor)}{typeArgs}) _this).{FieldName(f, i)}"
+            : $"_this.{DestructorGetterName(f, ctor, i)}"
+          : FieldName(f, i);
```

Comment:
I find nested tertiary operators hard to read, I think it's worth making this a few separate statements.

Reply:
It's indented and everything.